hikari/api/rest.py

```diff
@@ -507,7 +480,7 @@ async def edit_permission_overwrites(
             snowflakes.Snowflakeish, users.PartialUser, guilds.PartialRole, channels_.PermissionOverwrite
         ],
         *,
-        target_type: undefined.UndefinedOr[channels_.PermissionOverwriteType] = undefined.UNDEFINED,
+        target_type: undefined.UndefinedOr[typing.Union[channels_.PermissionOverwriteType, str]] = undefined.UNDEFINED,
```

Comment:
The implementation's typing for this needs to be updated as well, along with the type in the docstring.

Reply:
This was edited to match the implementation type, which it didnt match. It is true that the typehint has to be updated as well

Reply:
the too many unions behaviour seems to go away if that's correctly typed as [int | PermissionOverwriteType] and this does look a bit geh but if you assert that the expected different signatures are matched like the following then mypy doesn't complain 
```py
        if target_type is undefined.UNDEFINED:
            assert not isinstance(target, int)
            return await self.app.rest.edit_permission_overwrites(
                self.id, target, allow=allow, deny=deny, reason=reason
            )

        assert isinstance(target, int)
        return await self.app.rest.edit_permission_overwrites(
            self.id, target, target_type=target_type, allow=allow, deny=deny, reason=reason
        )
```